src/scenic/__main__.py

```diff
@@ -183,11 +196,17 @@ def runSimulation(scene):
                         break
             else:
                 if delay is None:
-                    scene.show(zoom=args.zoom)
+                    if mode_2d:
+                        scene.show_2d(zoom=args.zoom)
+                    else:
+                        scene.show(zoom=args.zoom)
                 else:
-                    scene.show(zoom=args.zoom, block=False)
-                    plt.pause(delay)
-                    plt.clf()
+                    if mode_2d:
+                        scene.show(zoom=args.zoom, block=False)
+                        plt.pause(delay)
+                        plt.clf()
+                    else:
+                        scene.show(zoom=args.zoom)
```

Comment:
This doesn't look right: where do you cause the window to be closed after `delay`? It looks like this will block in the same way it does when `delay` is `None`.

Reply:
Ah right, I haven't implemented this functionality yet. I'll look into doing it with `pyglet`.